docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md

```diff
@@ -0,0 +1,327 @@
+
+```json
+module "opensearch_role" {
+  source    = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  role_name = "opensearch-role"
+
+  role_policy_arns = {
+    policy = aws_iam_policy.opensearch_policy.arn
+  }
+
+  oidc_providers = {
+    main = {
+      provider_arn               = "arn:aws:iam::account-id:oidc-provider/oidc.eks.region.amazonaws.com/id/eks-id"
+      namespace_service_accounts = ["opensearch-namespace:opensearch-serviceaccount"]
+    }
+  }
+}
+```
+
+These two Terraform snippets will allow the service account `opensearch-serviceaccount` within the `opensearch-namespace` to generally access the AWS OpenSearch service for the `test-domain` cluster.
+
+The output of the module `opensearch_role` has the output `iam_role_arn` to annotate a service account to use the mapping.
+
+Annotate the service account with the `iam_role_arn` output of the `opensearch_role`.
+
+```
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    eks.amazonaws.com/role-arn: arn:aws:iam::account-id:role/role-name
+  name: opensearch-serviceaccount
+  namespace: opensearch-namespace
+```
+
+This setup is sufficient for OpenSearch clusters without `fine-grained access control`.
+
+`Fine-grained access control` adds another layer of security to OpenSearch, requiring you to add a mapping between the IAM role and the internal OpenSearch role. Visit the [AWS documentation](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/fgac.html) on `fine-grained access control`.
+
+There are different ways to configure the mapping within OpenSearch:
+
+- Via a [Terraform module](https://registry.terraform.io/modules/idealo/opensearch/aws/latest) in case your OpenSearch instance is exposed.
+- Via the [OpenSearch dashboard](https://opensearch.org/docs/latest/security/access-control/users-roles/).
+- Via the REST API.
+
+The important part is assigning the `iam_role_arn` of the previously created `opensearch_role` to an internal role within OpenSearch. For example, `all_access` on the OpenSearch side is a good candidate, or if required, extra roles can be created with more restrictive access.
+
+### Operate
+
+Configure Operate to use the feature set of IRSA for the OpenSearch Exporter. See the configuration reference [here](../../../../operate-deployment/operate-configuration.md#elasticsearch-or-opensearch).
+
+As an example, configure the following environment variables:
+
+```
+- name: CAMUNDA_OPERATE_ELASTICSEARCH_URL
+  value: https://test-domain.region.es.amazonaws.com
+- name: CAMUNDA_OPERATE_ZEEBEELASTICSEARCH_URL
+  value: https://test-domain.region.es.amazonaws.com
+```
+
+Where the value is whatever the endpoint of your OpenSearch cluster is.
+
+:::note
+AWS OpenSearch runs generally on port 443.
```

Comment:
I don't know if it's only me, but for a second, the word `run` confused me even its technically valid :sweat_smile: 
Maybe "listen" more a clear word :thinking:

Reply:
rephrased it to `listens` and added that it usually runs on 9200, so people may better understand why it's noted.